.github/workflows/python.yml

```
@@ -2,25 +2,43 @@ name: Python CI
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
-    branches: [ main ]
+    branches: [main]
 
 jobs:
-  build:
-
+  test:
+    name: Tests
     runs-on: ubuntu-latest
     strategy:
       max-parallel: 4
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
+        django-version: ["3.2", "4.2", "5.2"]
+        exclude:
+          # Django 3.2 doesn't support Python 3.12+
+          - python-version: "3.12"
+            django-version: "3.2"
+          - python-version: "3.13"
+            django-version: "3.2"
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Run Tests
-      run: |
-        python setup.py test
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ "${{ matrix.django-version }}" == "3.2" ]; then
+            pip install "django>=3.2,<4.0"
+          elif [ "${{ matrix.django-version }}" == "4.2" ]; then
+            pip install "django>=4.2,<5.0"
+          else
+            pip install "django>=5.2,<6.0"
+          fi
```

why not trust the matrix exclude and use ${{ matrix.django-version }} variable to select the compatible django version?

```suggestion
          pip install "django==${{ matrix.django-version }}"
```